### PR TITLE
select ebv dataset via url param

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Man muss den EBV-Pfad via URL-Parameter entgegennehmen](https://trello.com/c/ftK07HcR/583-man-muss-den-ebv-pfad-via-url-parameter-entgegennehmen)

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts [hunk 1]
```diff
@@ -152,6 +156,39 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
...
+                mergeMap(([params, sessionToken]) =>
+                    this.http.get<EbvDataset>(`${this.config.API_URL}/ebv/dataset/${params.id}`, {
+                        headers: new HttpHeaders().set('Authorization', `Bearer ${sessionToken}`),
+                    }),
+                ),
+            )
+            .subscribe((dataset) => {
+                const ebvClass = this.ebvClasses.find((c) => c.name === dataset.ebvClass);
+
+                if (!!ebvClass) {
```
Comment (ChristianBeilschmidt) [inline projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts:173]: Why can't you just say
```suggestion
                if (ebvClass) {
```
?

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts [hunk 2]
```diff
@@ -152,6 +156,39 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
...
+                    this.ebvNames = ebvClass.ebvNames;
+
+                    this.ebvName = dataset.ebvName;
+
+                    this.request<Array<EbvDataset>>(`ebv/datasets/${encodeURIComponent(dataset.ebvName)}`, undefined, (data) => {
+                        this.ebvDatasets = data;
+
+                        const selected = data.find((d) => d.id === dataset.id);
+
+                        if (!!selected) {
```
Comment (ChristianBeilschmidt) [inline projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts:184]: same question here
```suggestion
                        if (selected) {
```

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [inline projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts:173]: You could use some early returns in order to remove two of the four layers of hierarchy in this method.

Despite that, should we show some notification when this went wrong or do we silently ignore it?

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts [hunk 3]
```diff
@@ -152,6 +156,39 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
...
+                            this.setEbvDataset(selected);
```
Comment (ChristianBeilschmidt) [inline projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts:185]: So, this does not re-assembly https://portal.geobon.org/map/#/?id=1 but stops at the dataset level?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts (hunk 1 above)
Comment (michaelmattig) [reply projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts]: efbfc5f59b61d6e75ff375d0188651c370a846cb

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts (hunk 2 above)
Comment (michaelmattig) [reply projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts]: efbfc5f59b61d6e75ff375d0188651c370a846cb

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts (hunk 3 above)
Comment (michaelmattig) [reply projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts]: efbfc5f59b61d6e75ff375d0188651c370a846cb

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts [hunk 4]
```diff
@@ -152,6 +164,59 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
...
+        let groups = this.ebvTree.tree.groups;
+        let index = 0;
+
+        while (groups.length > 0) {
+            this.setEbvPath(groups[0], index++);
+            groups = groups[0].groups;
+        }
+
+        const entity = this.ebvTree.tree.entities[0];
+        this.setEbvEntity(entity);
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts:217]: 9e9a7173699b1158d5dd6b6cf8d6ee90712e288d

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**